orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/ExecutionRunner.java

```diff
@@ -46,8 +49,9 @@ default void cancel(
   }
 
   class NoSuchStageDefinitionBuilder extends RuntimeException {
-    public NoSuchStageDefinitionBuilder(String type) {
-      super(format("No StageDefinitionBuilder implementation for %s found", type));
+    public NoSuchStageDefinitionBuilder(String type, Collection<String> knownTypes) {
+      super(format("No StageDefinitionBuilder implementation for %s found.%s", type,
+        knownTypes == null || knownTypes.size() == 0 ? "" : format(" Known stage types: %s", String.join(",", knownTypes))));
```

Comment:
In the off chance there are no builders found, it would be nice to include that info (e.g. ` (No stage types registered)`) instead of an empty string after the period.